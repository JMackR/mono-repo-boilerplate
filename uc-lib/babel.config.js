module.exports = {
  presets: [["@babel/preset-env", { targets: { node: "current" } }], "module:metro-react-native-babel-preset"],
  plugins: ["@babel/plugin-proposal-class-properties", "inline-json-import"],
  sourceType: "unambiguous",
}
