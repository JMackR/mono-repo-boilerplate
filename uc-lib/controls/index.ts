export * from "./image"
export * from "./button"
export * from "./text"
export * from "./activity-indicator"
export * from "./separator"
export * from "./list"
export * from "./section-list"
export * from "./badge"
export * from "./toggle"
export * from "./login-button"
export * from "./layout"
export * from "./background"
export * from "./grid"
export * from "./touchable"
export * from "./view-block"
export * from "./text-entry"
export * from "./scroll-view"
export * from "./circle"
export * from "./slider"
export * from "./clickable"
export * from "./map"
export * from "./visibility-check"
export * from "./strike-through-text"
// export * from "./stripe-text-entry"
export * from "./input"
