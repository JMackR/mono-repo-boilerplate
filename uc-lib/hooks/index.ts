export * from "./use-infinite-scroll"
export * from "./selectable"
export * from "./screen-provider"
export * from "./loading"
