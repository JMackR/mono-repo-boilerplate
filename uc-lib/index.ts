export * from "./animations"
export * from "./assets"
export * from "./controls"
export * from "./themes"
export * from "./widgets"
export * from "./utilities"
export * from "./hooks"
export * from "./validators"
