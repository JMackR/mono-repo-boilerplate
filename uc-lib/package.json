{
  "name": "redibs-ucl",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "test": "jest --projects test/react test/react-native test/non-react"
  },
  "dependencies": {
    "@types/ramda": "^0.27.14",
    "classnames": "2.2.6",
    "google-map-react": "1.1.7",
    "hoist-non-react-statics": "3.3.2",
    "imask": "6.0.5",
    "react-compound-slider": "2.5.0",
    "react-native-autolink": "3.0.0",
    "react-native-fs": "^2.16.6",
    "zxcvbn": "4.4.2"
  },
  "devDependencies": {},
  "prettier": {
    "printWidth": 125,
    "semi": false,
    "singleQuote": false,
    "trailingComma": "all"
  },
  "jest-junit": {
    "outputDirectory": "../build",
    "ancestorSeparator": " â€º ",
    "classNameTemplate": "{classname}",
    "suiteName": "redibs-redibs-ucl jest tests",
    "suiteNameTemplate": "{filename}",
    "titleTemplate": "{title}",
    "uniqueOutputName": "true"
  }
}
