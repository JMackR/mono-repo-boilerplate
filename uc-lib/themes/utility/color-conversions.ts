// tslint:disable: no-magic-numbers
// tslint:disable: no-bitwise
export const ensureHex = (input: string): string => {
  if (input.startsWith("rgba")) {
    return rgba2hex(input, input.length === 9).toUpperCase()
  } else if (!input.startsWith("#")) {
    // tslint:disable-next-line: no-console
    console.warn("unable to ensure hex", input)
  }
  return input.toUpperCase()
}

/**
 * Adapted from https://stackoverflow.com/questions/49974145/how-to-convert-rgba-to-hex-color-code-using-javascript#49974627
 * @param orig
 * @param forceAlpha
 */
function rgba2hex(orig: string, forceAlpha?: boolean) {
  let a
  const rgb = orig.replace(/\s/g, "").match(/^rgba?\((\d+),(\d+),(\d+),?([^,\s)]+)?/i)
  const alpha = ((rgb && rgb[4]) || "").trim()
  let hex = rgb
    ? (rgb[1] | (1 << 8)).toString(16).slice(1) +
      (rgb[2] | (1 << 8)).toString(16).slice(1) +
      (rgb[3] | (1 << 8)).toString(16).slice(1)
    : orig

  if (alpha !== "") {
    a = alpha
  } else {
    a = 1
  }
  // multiply before convert to HEX
  a = ((a * 255) | (1 << 8)).toString(16).slice(1)
  hex = hex

  if (a !== "ff" || forceAlpha === true) {
    return "#" + a + hex
  } else if (a === "ff") {
    return "#" + hex
  }
  throw Error("unexpected rgba2hex() conversion!")
}
