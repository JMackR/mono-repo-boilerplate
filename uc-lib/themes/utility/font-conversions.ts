import { FontStyle } from "../type-defs"

import { StyleSheet, CSSPropertiesComplete } from "aphrodite/no-important"
/**
 * Converts the FontStyle to an aphrodite StyleSheet object
 * @param font
 */
export const covertFontToWebStyleSheet = (
  font: FontStyle,
  color?: string,
  display: CSSPropertiesComplete["display"] = "inline",
) => {
  const fontStyles = StyleSheet.create({
    text: {
      color,
      fontSize: font.fontSize,
      fontWeight: convertFontWeight(font.fontWeight),
      fontFamily: font.fontFamily + ", sans-serif",
      display,
    },
  })
  return fontStyles
}

/**
 * Convert a font weight for use with aphrodite for web
 * @param weight
 */
// tslint:disable: no-magic-numbers
export const convertFontWeight = (
  weight: string,
):
  | "initial"
  | "inherit"
  | "unset"
  | "normal"
  | "bold"
  | "bolder"
  | "lighter"
  | 100
  | 200
  | 300
  | 400
  | 500
  | 600
  | 700
  | 800
  | 900
  | undefined => {
  switch (weight) {
    case "100":
      return 100
    case "200":
      return 200
    case "300":
      return 300
    case "400":
      return 400
    case "500":
      return 500
    case "600":
      return 600
    case "700":
      return 700
    case "800":
      return 800
    case "900":
      return 900
    case "initial":
      return "initial"
    case "inherit":
      return "inherit"
    case "unset":
      return "unset"
    case "normal":
      return "normal"
    case "bold":
      return "bold"
    case "bolder":
      return "bolder"
    case "lighter":
      return "lighter"
    default:
      return undefined
  }
}

/**
 * Converts an FontTheme line height for use with web.
 * @param lineHeight line height from an FontTheme.
 */
export const convertLineHeightToPx = (lineHeight: number): string => {
  return lineHeight + "px"
}
