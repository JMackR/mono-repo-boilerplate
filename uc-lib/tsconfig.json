{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "baseUrl": ".",
    "outDir": "lib",
    "rootDir": "../",
    "jsx": "react-native",
    "composite": true,
    "declarationMap": true,
    "isolatedModules": true,
    "sourceMap": true,
    "moduleResolution": "Node",
    "esModuleInterop": true,
    "typeRoots": ["../node_modules/@types", "./types"]
  },
  "include": ["./"],
  "exclude": ["node_modules", "babel.config.js", "metro.config.js", "jest.config.js", "types"]
}
