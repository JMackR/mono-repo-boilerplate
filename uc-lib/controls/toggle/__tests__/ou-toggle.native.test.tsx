import React from "react"
import { shallow, ShallowWrapper } from "enzyme"
import { Toggle } from "uc-lib"
import { render } from "react-native-testing-library"
import { ouMockPlatform, ouResetPlatformMock } from "uc-lib/test/react-native/utility"

describe("Toggle Functionality Tests", () => {
  let wrapper: ShallowWrapper
  const click = jest.fn()
  beforeEach(() => {
    wrapper = shallow(<Toggle state={false} onChange={click} />)
  })
  test("Click calls passed in onChange function", () => {
    wrapper.props().onValueChange(true)
    expect(click).toHaveBeenCalledTimes(1)
  })
})

describe("Toggle Snapshot Tests", () => {
  test("Toggle with on state renderer.creates properly", () => {
    const toggle = render(<Toggle state={true} onChange={jest.fn()} />)
    expect(toggle).toMatchSnapshot()
  })
  test("Toggle with off state renderer.creates properly", () => {
    const toggle = render(<Toggle state={false} onChange={jest.fn()} />)
    expect(toggle).toMatchSnapshot()
  })
  test("Toggle with disabled state renderer.creates properly", () => {
    const toggle = render(<Toggle state={false} onChange={jest.fn()} disabled={true} />)
    expect(toggle).toMatchSnapshot()
  })
  describe("Android Snapshot", () => {
    ouMockPlatform("android")
    test("Android toggle with state renderer.creates properly", () => {
      const toggle = render(<Toggle state={false} onChange={jest.fn()} />)
      expect(toggle).toMatchSnapshot()
    })
    ouResetPlatformMock()
  })
  describe("iOS Snapshot", () => {
    ouMockPlatform("ios")
    test("iOS toggle with state renderer.creates properly", () => {
      const toggle = render(<Toggle state={false} onChange={jest.fn()} />)
      expect(toggle).toMatchSnapshot()
    })
    ouResetPlatformMock()
  })
})
