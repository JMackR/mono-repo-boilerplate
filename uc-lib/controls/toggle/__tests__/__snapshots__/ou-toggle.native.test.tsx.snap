// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle Snapshot Tests Android Snapshot Android toggle with state renderer.creates properly 1`] = `
<AndroidSwitch
  accessibilityLabel="redibs-ucl.toggle"
  accessibilityRole="button"
  enabled={true}
  on={false}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="redibs-ucl.toggle"
  thumbTintColor="#E2E2E2"
  trackColorForFalse="#ABABAB"
  trackColorForTrue="#00A87E"
  trackTintColor="#ABABAB"
/>
`;

exports[`Toggle Snapshot Tests Toggle with disabled state renderer.creates properly 1`] = `
<RCTSwitch
  accessibilityLabel="redibs-ucl.toggle"
  accessibilityRole="button"
  disabled={true}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="#00A87E"
  style={
    Array [
      Object {
        "height": 31,
        "width": 51,
      },
      Array [
        Object {
          "borderColor": "#ABABAB",
          "borderWidth": 0.5,
        },
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 16,
        },
      ],
    ]
  }
  testID="redibs-ucl.toggle"
  tintColor="#FFFFFF"
  value={false}
/>
`;

exports[`Toggle Snapshot Tests Toggle with off state renderer.creates properly 1`] = `
<RCTSwitch
  accessibilityLabel="redibs-ucl.toggle"
  accessibilityRole="button"
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="#00A87E"
  style={
    Array [
      Object {
        "height": 31,
        "width": 51,
      },
      Array [
        Object {
          "borderColor": "#ABABAB",
          "borderWidth": 0.5,
        },
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 16,
        },
      ],
    ]
  }
  testID="redibs-ucl.toggle"
  tintColor="#FFFFFF"
  value={false}
/>
`;

exports[`Toggle Snapshot Tests Toggle with on state renderer.creates properly 1`] = `
<RCTSwitch
  accessibilityLabel="redibs-ucl.toggle"
  accessibilityRole="button"
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="#00A87E"
  style={
    Array [
      Object {
        "height": 31,
        "width": 51,
      },
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 16,
      },
    ]
  }
  testID="redibs-ucl.toggle"
  tintColor="#FFFFFF"
  value={true}
/>
`;

exports[`Toggle Snapshot Tests iOS Snapshot iOS toggle with state renderer.creates properly 1`] = `
<RCTSwitch
  accessibilityLabel="redibs-ucl.toggle"
  accessibilityRole="button"
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="#00A87E"
  style={
    Array [
      Object {
        "height": 31,
        "width": 51,
      },
      Array [
        Object {
          "borderColor": "#ABABAB",
          "borderWidth": 0.5,
        },
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 16,
        },
      ],
    ]
  }
  testID="redibs-ucl.toggle"
  tintColor="#FFFFFF"
  value={false}
/>
`;
